File: payment/src/main/java/com/trinhvu/payment/viewmodel/CapturePayment.java
package com.trinhvu.payment.viewmodel;

import com.trinhvu.payment.model.enumeration.PaymentMethod;
import com.trinhvu.payment.model.enumeration.PaymentStatus;

import java.math.BigDecimal;

public record CapturePayment(
        Long orderId,
        Long checkOutId,
        BigDecimal amount,
        PaymentMethod paymentMethod,
        PaymentStatus paymentStatus,
        String failureMessage
) {
}

File: stock-watch-fe/components/payment/models/CapturePayment.ts
export interface CapturePayment {
  orderId: number;
  checkOutId: string;
  amount: number; // Or 'BigDecimal' if using a BigDecimal library
  paymentMethod: string; // Define this enum separately
  paymentStatus: string; // Define this enum separately
  failureMessage: string;
}
